sdl-core/src/main/scala/io/smartdatalake/workflow/ActionPipelineContext.scala

```diff
@@ -54,19 +54,21 @@ import scala.collection.mutable
  */
 @DeveloperApi
 case class ActionPipelineContext (
-                                  feed: String, application: String, executionId: SDLExecutionId,
-                                  instanceRegistry: InstanceRegistry,
-                                  referenceTimestamp: Option[LocalDateTime] = None,
-                                  appConfig: SmartDataLakeBuilderConfig, // application config is needed to persist action dag state for recovery
-                                  runStartTime: LocalDateTime = LocalDateTime.now(),
-                                  attemptStartTime: LocalDateTime = LocalDateTime.now(),
-                                  simulation: Boolean = false,
-                                  var phase: ExecutionPhase = ExecutionPhase.Prepare,
-                                  dataFrameReuseStatistics: mutable.Map[(DataObjectId, Seq[PartitionValues]), Seq[ActionId]] = mutable.Map(),
-                                  actionsSelected: Seq[ActionId] = Seq(),
-                                  actionsSkipped: Seq[ActionId] = Seq(),
-                                  serializableHadoopConf: SerializableHadoopConfiguration
-) extends SmartDataLakeLogger {
+                                   feed: String, application: String, executionId: SDLExecutionId,
+                                   @transient
+                                   instanceRegistry: InstanceRegistry,
+                                   referenceTimestamp: Option[LocalDateTime] = None,
+                                   appConfig: SmartDataLakeBuilderConfig, // application config is needed to persist action dag state for recovery
+                                   runStartTime: LocalDateTime = LocalDateTime.now(),
+                                   attemptStartTime: LocalDateTime = LocalDateTime.now(),
+                                   simulation: Boolean = false,
+                                   var phase: ExecutionPhase = ExecutionPhase.Prepare,
+                                   dataFrameReuseStatistics: mutable.Map[(DataObjectId, Seq[PartitionValues]), Seq[ActionId]] = mutable.Map(),
+                                   actionsSelected: Seq[ActionId] = Seq(),
+                                   actionsSkipped: Seq[ActionId] = Seq(),
+                                   @transient
```

Comment:
Did you have problems with serializing SerializableHadoopConfiguration? That would be strange as it is explicitly a wrapper to make HadoopConfiguration serializable...

Reply:
You are right, the serilization works. It just looked a bit noisy to me, but probably it does display what it shoud.
This is how the output looks like without the transient annotation:

`{"feed":".*","application":".*","executionId":{"runId":1,"attemptId":1},"referenceTimestamp":[2022,2,7,14,58,37,62000000],"appConfig":{"feedSel":".*","applicationName":null,"configuration":null,"master":"local[*]","deployMode":"client","username":null,"kerberosDomain":null,"keytabPath":null,"partitionValues":null,"multiPartitionValues":null,"parallelism":1,"statePath":null,"overrideJars":null,"test":null,"streaming":false,"appName":".*","dryRun":false},"runStartTime":[2022,2,7,14,58,34,314000000],"attemptStartTime":[2022,2,7,14,58,34,315000000],"simulation":false,"phase":{"enumClass":"io.smartdatalake.workflow.ExecutionPhase","value":"Init"},"dataFrameReuseStatistics":{},"actionsSelected":[{"id":"select-some-cols"}],"actionsSkipped":[],"serializableHadoopConf":{"conf":{"finalParameters":["mapreduce.job.end-notification.max.retry.interval","mapreduce.job.end-notification.max.attempts"],"classLoader":{"parent":{"parent":null,"urls":["file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/ext/access-bridge-64.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/ext/cldrdata.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/ext/dnsns.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/ext/jaccess.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/ext/jfxrt.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/ext/localedata.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/ext/nashorn.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/ext/sunec.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/ext/sunjce_provider.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/ext/sunmscapi.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/ext/sunpkcs11.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/ext/zipfs.jar"]},"urls":["file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/charsets.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/deploy.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/ext/access-bridge-64.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/ext/cldrdata.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/ext/dnsns.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/ext/jaccess.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/ext/jfxrt.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/ext/localedata.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/ext/nashorn.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/ext/sunec.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/ext/sunjce_provider.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/ext/sunmscapi.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/ext/sunpkcs11.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/ext/zipfs.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/javaws.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/jce.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/jfr.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/jfxswt.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/jsse.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/management-agent.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/plugin.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/resources.jar","file:/C:/Program%20Files/Java/jdk1.8.0_172/jre/lib/rt.jar","file:/C:/Users/tbb/WORK/sdl/smart-data-lake/sdl-core/target/classes/","file:/C:/Users/tbb/.ivy2/cache/org.scala-lang/scala-library/jars/scala-library-2.12.12.jar","file:/C:/Users/tbb/.ivy2/cache/org.scala-lang/scala-reflect/jars/scala-reflect-2.12.12.jar","file:/C:/Users/tbb/.ivy2/cache/org.scala-lang.modules/scala-xml_2.12/bundles/scala-xml_2.12-1.0.6.jar","file:/C:/Users/tbb/.m2/repository/com/databricks/dbutils-api_2.11/0.0.4/dbutils-api_2.11-0.0.4.jar","file:/C:/Users/tbb/.m2/repository/org/slf4j/slf4j-api/1.7.30/slf4j-api-1.7.30.jar","file:/C:/Users/tbb/.m2/repository/com/github/scopt/scopt_2.12/4.0.1/scopt_2.12-4.0.1.jar","file:/C:/Users/tbb/.m2/repository/org/apache/spark/spark-core_2.12/3.2.0/spark-core_2.12-3.2.0.jar","file:/C:/Users/tbb/.m2/repository/org/apache/avro/avro/1.10.2/avro-1.10.2.jar","file:/C:/Users/tbb/.m2/repository/org/apache/avro/avro-mapred/1.10.2/avro-mapred-1.10.2.jar","file:/C:/Users/tbb/.m2/repository/org/apache/avro/avro-ipc/1.10.2/avro-ipc-1.10.2.jar","file:/C:/Users/tbb/.m2/repository/com/twitter/chill_2.12/0.10.0/chill_2.12-0.10.0.jar","file:/C:/Users/tbb/.m2/repository/com/esotericsoftware/kryo-shaded/4.0.2/kryo-shaded-4.0.2.jar","file:/C:/Users/tbb/.m2/repository/com/esotericsoftware/minlog/1.3.0/minlog-1.3.0.jar","file:/C:/Users/tbb/.m2/repository/org/objenesis/objenesis/2.5.1/objenesis-2.5.1.jar","file:/C:/Users/tbb/.m2/repository/com/twitter/chill-java/0.10.0/chill-java-0.10.0.jar","file:/C:/Users/tbb/.m2/repository/org/apache/xbean/xbean-asm9-shaded/4.20/xbean-asm9-shaded-4.20.jar","file:/C:/Users/tbb/.m2/repository/org/apache/hadoop/hadoop-client-api/3.3.1/hadoop-client-api-3.3.1.jar","file:/C:/Users/tbb/.m2/repository/org/apache/hadoop/hadoop-client-runtime/3.3.1/hadoop-client-runtime-3.3.1.jar","file:/C:/Users/tbb/.m2/repository/org/apache/htrace/htrace-core4/4.1.0-incubating/htrace-core4-4.1.0-incubating.jar","file:/C:/Users/tbb/.m2/repository/commons-logging/commons-logging/1.1.3/commons-logging-1.1.3.jar","file:/C:/Users/tbb/.m2/repository/org/apache/spark/spark-launcher_2.12/3.2.0/spark-launcher_2.12-3.2.0.jar","file:/C:/Users/tbb/.m2/repository/org/apache/spark/spark-kvstore_2.12/3.2.0/spark-kvstore_2.12-3.2.0.jar","file:/C:/Users/tbb/.m2/repository/org/fusesource/leveldbjni/leveldbjni-all/1.8/leveldbjni-all-1.8.jar","file:/C:/Users/tbb/.m2/repository/org/apache/spark/spark-network-common_2.12/3.2.0/spark-network-common_2.12-3.2.0.jar","file:/C:/Users/tbb/.m2/repository/com/google/crypto/tink/tink/1.6.0/tink-1.6.0.jar","file:/C:/Users/tbb/.m2/repository/org/apache/spark/spark-network-shuffle_2.12/3.2.0/spark-network-shuffle_2.12-3.2.0.jar","file:/C:/Users/tbb/.m2/repository/org/apache/spark/spark-unsafe_2.12/3.2.0/spark-unsafe_2.12-3.2.0.jar","file:/C:/Users/tbb/.m2/repository/javax/activation/activation/1.1.1/activation-1.1.1.jar","file:/C:/Users/tbb/.m2/repository/org/apache/curator/curator-recipes/2.13.0/curator-recipes-2.13.0.jar","file:/C:/Users/tbb/.m2/repository/org/apache/curator/curator-framework/2.13.0/curator-framework-2.13.0.jar","file:/C:/Users/tbb/.m2/repository/org/apache/curator/curator-client/2.13.0/curator-client-2.13.0.jar","file:/C:/Users/tbb/.m2/repository/org/apache/zookeeper/zookeeper/3.6.2/zookeeper-3.6.2.jar","file:/C:/Users/tbb/.m2/repository/commons-lang/commons-lang/2.6/commons-lang-2.6.jar","file:/C:/Users/tbb/.m2/repository/org/apache/zookeeper/zookeeper-jute/3.6.2/zookeeper-jute-3.6.2.jar","file:/C:/Users/tbb/.m2/repository/org/apache/yetus/audience-annotations/0.5.0/audience-annotations-0.5.0.jar","file:/C:/Users/tbb/.m2/repository/jakarta/servlet/jakarta.servlet-api/4.0.3/jakarta.servlet-api-4.0.3.jar","file:/C:/Users/tbb/.m2/repository/commons-codec/commons-codec/1.15/commons-codec-1.15.jar","file:/C:/Users/tbb/.m2/repository/org/apache/commons/commons-lang3/3.12.0/commons-lang3-3.12.0.jar","file:/C:/Users/tbb/.m2/repository/org/apache/commons/commons-math3/3.4.1/commons-math3-3.4.1.jar","file:/C:/Users/tbb/.m2/repository/org/apache/commons/commons-text/1.6/commons-text-1.6.jar","file:/C:/Users/tbb/.m2/repository/commons-collections/commons-collections/3.2.2/commons-collections-3.2.2.jar","file:/C:/Users/tbb/.m2/repository/com/google/code/findbugs/jsr305/3.0.0/jsr305-3.0.0.jar","file:/C:/Users/tbb/.m2/repository/org/slf4j/jul-to-slf4j/1.7.30/jul-to-slf4j-1.7.30.jar","file:/C:/Users/tbb/.m2/repository/org/slf4j/jcl-over-slf4j/1.7.30/jcl-over-slf4j-1.7.30.jar","file:/C:/Users/tbb/.m2/repository/org/slf4j/slf4j-log4j12/1.7.30/slf4j-log4j12-1.7.30.jar","file:/C:/Users/tbb/.m2/repository/com/ning/compress-lzf/1.0.3/compress-lzf-1.0.3.jar","file:/C:/Users/tbb/.m2/repository/org/xerial/snappy/snappy-java/1.1.8.4/snappy-java-1.1.8.4.jar","file:/C:/Users/tbb/.m2/repository/org/lz4/lz4-java/1.7.1/lz4-java-1.7.1.jar","file:/C:/Users/tbb/.m2/repository/com/github/luben/zstd-jni/1.5.0-4/zstd-jni-1.5.0-4.jar","file:/C:/Users/tbb/.m2/repository/org/roaringbitmap/RoaringBitmap/0.7.45/RoaringBitmap-0.7.45.jar","file:/C:/Users/tbb/.m2/repository/org/roaringbitmap/shims/0.7.45/shims-0.7.45.jar","file:/C:/Users/tbb/.m2/repository/commons-net/commons-net/3.1/commons-net-3.1.jar","file:/C:/Users/tbb/.m2/repository/org/json4s/json4s-jackson_2.12/3.7.0-M11/json4s-jackson_2.12-3.7.0-M11.jar","file:/C:/Users/tbb/.m2/repository/org/json4s/json4s-core_2.12/3.7.0-M11/json4s-core_2.12-3.7.0-M11.jar","file:/C:/Users/tbb/.m2/repository/org/json4s/json4s-ast_2.12/3.7.0-M11/json4s-ast_2.12-3.7.0-M11.jar","file:/C:/Users/tbb/.m2/repository/org/json4s/json4s-scalap_2.12/3.7.0-M11/json4s-scalap_2.12-3.7.0-M11.jar","file:/C:/Users/tbb/.m2/repository/org/glassfish/jersey/core/jersey-client/2.34/jersey-client-2.34.jar","file:/C:/Users/tbb/.m2/repository/jakarta/ws/rs/jakarta.ws.rs-api/2.1.6/jakarta.ws.rs-api-2.1.6.jar","file:/C:/Users/tbb/.m2/repository/org/glassfish/hk2/external/jakarta.inject/2.6.1/jakarta.inject-2.6.1.jar","file:/C:/Users/tbb/.m2/repository/org/glassfish/jersey/core/jersey-common/2.34/jersey-common-2.34.jar","file:/C:/Users/tbb/.m2/repository/jakarta/annotation/jakarta.annotation-api/1.3.5/jakarta.annotation-api-1.3.5.jar","file:/C:/Users/tbb/.m2/repository/org/glassfish/hk2/osgi-resource-locator/1.0.3/osgi-resource-locator-1.0.3.jar","file:/C:/Users/tbb/.m2/repository/org/glassfish/jersey/core/jersey-server/2.34/jersey-server-2.34.jar","file:/C:/Users/tbb/.m2/repository/jakarta/validation/jakarta.validation-api/2.0.2/jakarta.validation-api-2.0.2.jar","file:/C:/Users/tbb/.m2/repository/org/glassfish/jersey/containers/jersey-container-servlet/2.34/jersey-container-servlet-2.34.jar","file:/C:/Users/tbb/.m2/repository/org/glassfish/jersey/containers/jersey-container-servlet-core/2.34/jersey-container-servlet-core-2.34.jar","file:/C:/Users/tbb/.m2/repository/org/glassfish/jersey/inject/jersey-hk2/2.34/jersey-hk2-2.34.jar","file:/C:/Users/tbb/.m2/repository/org/glassfish/hk2/hk2-locator/2.6.1/hk2-locator-2.6.1.jar","file:/C:/Users/tbb/.m2/repository/org/glassfish/hk2/external/aopalliance-repackaged/2.6.1/aopalliance-repackaged-2.6.1.jar","file:/C:/Users/tbb/.m2/repository/org/glassfish/hk2/hk2-api/2.6.1/hk2-api-2.6.1.jar","file:/C:/Users/tbb/.m2/repository/org/glassfish/hk2/hk2-utils/2.6.1/hk2-utils-2.6.1.jar","file:/C:/Users/tbb/.m2/repository/io/netty/netty-all/4.1.68.Final/netty-all-4.1.68.Final.jar","file:/C:/Users/tbb/.m2/repository/com/clearspring/analytics/stream/2.9.6/stream-2.9.6.jar","file:/C:/Users/tbb/.m2/repository/io/dropwizard/metrics/metrics-core/4.2.0/metrics-core-4.2.0.jar","file:/C:/Users/tbb/.m2/repository/io/dropwizard/metrics/metrics-jvm/4.2.0/metrics-jvm-4.2.0.jar","file:/C:/Users/tbb/.m2/repository/io/dropwizard/metrics/metrics-json/4.2.0/metrics-json-4.2.0.jar","file:/C:/Users/tbb/.m2/repository/io/dropwizard/metrics/metrics-graphite/4.2.0/metrics-graphite-4.2.0.jar","file:/C:/Users/tbb/.m2/repository/io/dropwizard/metrics/metrics-jmx/4.2.0/metrics-jmx-4.2.0.jar","file:/C:/Users/tbb/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.12.3/jackson-databind-2.12.3.jar","file:/C:/Users/tbb/.m2/repository/com/fasterxml/jackson/module/jackson-module-scala_2.12/2.12.3/jackson-module-scala_2.12-2.12.3.jar","file:/C:/Users/tbb/.m2/repository/com/thoughtworks/paranamer/paranamer/2.8/paranamer-2.8.jar","file:/C:/Users/tbb/.m2/repository/org/apache/ivy/ivy/2.5.0/ivy-2.5.0.jar","file:/C:/Users/tbb/.m2/repository/oro/oro/2.0.8/oro-2.0.8.jar","file:/C:/Users/tbb/.m2/repository/net/razorvine/pyrolite/4.30/pyrolite-4.30.jar","file:/C:/Users/tbb/.m2/repository/net/sf/py4j/py4j/0.10.9.2/py4j-0.10.9.2.jar","file:/C:/Users/tbb/.m2/repository/org/apache/commons/commons-crypto/1.1.0/commons-crypto-1.1.0.jar","file:/C:/Users/tbb/.m2/repository/org/spark-project/spark/unused/1.0.0/unused-1.0.0.jar","file:/C:/Users/tbb/.m2/repository/org/apache/spark/spark-sql_2.12/3.2.0/spark-sql_2.12-3.2.0.jar","file:/C:/Users/tbb/.m2/repository/org/rocksdb/rocksdbjni/6.20.3/rocksdbjni-6.20.3.jar","file:/C:/Users/tbb/.m2/repository/com/univocity/univocity-parsers/2.9.1/univocity-parsers-2.9.1.jar","file:/C:/Users/tbb/.m2/repository/org/apache/spark/spark-sketch_2.12/3.2.0/spark-sketch_2.12-3.2.0.jar","file:/C:/Users/tbb/.m2/repository/org/apache/orc/orc-core/1.6.11/orc-core-1.6.11.jar","file:/C:/Users/tbb/.m2/repository/org/apache/orc/orc-shims/1.6.11/orc-shims-1.6.11.jar","file:/C:/Users/tbb/.m2/repository/com/google/protobuf/protobuf-java/2.5.0/protobuf-java-2.5.0.jar","file:/C:/Users/tbb/.m2/repository/io/airlift/aircompressor/0.21/aircompressor-0.21.jar","file:/C:/Users/tbb/.m2/repository/org/jetbrains/annotations/17.0.0/annotations-17.0.0.jar","file:/C:/Users/tbb/.m2/repository/org/threeten/threeten-extra/1.5.0/threeten-extra-1.5.0.jar","file:/C:/Users/tbb/.m2/repository/org/apache/orc/orc-mapreduce/1.6.11/orc-mapreduce-1.6.11.jar","file:/C:/Users/tbb/.m2/repository/org/apache/hive/hive-storage-api/2.7.2/hive-storage-api-2.7.2.jar","file:/C:/Users/tbb/.m2/repository/org/apache/parquet/parquet-column/1.12.1/parquet-column-1.12.1.jar","file:/C:/Users/tbb/.m2/repository/org/apache/parquet/parquet-common/1.12.1/parquet-common-1.12.1.jar","file:/C:/Users/tbb/.m2/repository/org/apache/parquet/parquet-encoding/1.12.1/parquet-encoding-1.12.1.jar","file:/C:/Users/tbb/.m2/repository/org/apache/parquet/parquet-hadoop/1.12.1/parquet-hadoop-1.12.1.jar","file:/C:/Users/tbb/.m2/repository/org/apache/parquet/parquet-format-structures/1.12.1/parquet-format-structures-1.12.1.jar","file:/C:/Users/tbb/.m2/repository/org/apache/parquet/parquet-jackson/1.12.1/parquet-jackson-1.12.1.jar","file:/C:/Users/tbb/.m2/repository/org/apache/spark/spark-hive_2.12/3.2.0/spark-hive_2.12-3.2.0.jar","file:/C:/Users/tbb/.m2/repository/org/apache/hive/hive-common/2.3.9/hive-common-2.3.9.jar","file:/C:/Users/tbb/.m2/repository/commons-cli/commons-cli/1.2/commons-cli-1.2.jar","file:/C:/Users/tbb/.m2/repository/jline/jline/2.14.6/jline-2.14.6.jar","file:/C:/Users/tbb/.m2/repository/com/tdunning/json/1.8/json-1.8.jar","file:/C:/Users/tbb/.m2/repository/com/github/joshelser/dropwizard-metrics-hadoop-metrics2-reporter/0.1.2/dropwizard-metrics-hadoop-metrics2-reporter-0.1.2.jar","file:/C:/Users/tbb/.m2/repository/org/apache/hive/hive-exec/2.3.9/hive-exec-2.3.9-core.jar","file:/C:/Users/tbb/.m2/repository/org/apache/hive/hive-vector-code-gen/2.3.9/hive-vector-code-gen-2.3.9.jar","file:/C:/Users/tbb/.m2/repository/org/apache/velocity/velocity/1.5/velocity-1.5.jar","file:/C:/Users/tbb/.m2/repository/org/antlr/antlr-runtime/3.4/antlr-runtime-3.4.jar","file:/C:/Users/tbb/.m2/repository/org/antlr/stringtemplate/3.2.1/stringtemplate-3.2.1.jar","file:/C:/Users/tbb/.m2/repository/antlr/antlr/2.7.7/antlr-2.7.7.jar","file:/C:/Users/tbb/.m2/repository/org/antlr/ST4/4.0.4/ST4-4.0.4.jar","file:/C:/Users/tbb/.m2/repository/com/google/code/gson/gson/2.2.4/gson-2.2.4.jar","file:/C:/Users/tbb/.m2/repository/stax/stax-api/1.0.1/stax-api-1.0.1.jar","file:/C:/Users/tbb/.m2/repository/org/apache/hive/hive-metastore/2.3.9/hive-metastore-2.3.9.jar","file:/C:/Users/tbb/.m2/repository/javolution/javolution/5.5.1/javolution-5.5.1.jar","file:/C:/Users/tbb/.m2/repository/com/jolbox/bonecp/0.8.0.RELEASE/bonecp-0.8.0.RELEASE.jar","file:/C:/Users/tbb/.m2/repository/com/zaxxer/HikariCP/2.5.1/HikariCP-2.5.1.jar","file:/C:/Users/tbb/.m2/repository/org/datanucleus/datanucleus-api-jdo/4.2.4/datanucleus-api-jdo-4.2.4.jar","file:/C:/Users/tbb/.m2/repository/org/datanucleus/datanucleus-rdbms/4.1.19/datanucleus-rdbms-4.1.19.jar","file:/C:/Users/tbb/.m2/repository/commons-pool/commons-pool/1.5.4/commons-pool-1.5.4.jar","file:/C:/Users/tbb/.m2/repository/commons-dbcp/commons-dbcp/1.4/commons-dbcp-1.4.jar","file:/C:/Users/tbb/.m2/repository/javax/jdo/jdo-api/3.0.1/jdo-api-3.0.1.jar","file:/C:/Users/tbb/.m2/repository/javax/transaction/jta/1.1/jta-1.1.jar","file:/C:/Users/tbb/.m2/repository/org/datanucleus/javax.jdo/3.2.0-m3/javax.jdo-3.2.0-m3.jar","file:/C:/Users/tbb/.m2/repository/javax/transaction/transaction-api/1.1/transaction-api-1.1.jar","file:/C:/Users/tbb/.m2/repository/org/apache/hive/hive-serde/2.3.9/hive-serde-2.3.9.jar","file:/C:/Users/tbb/.m2/repository/net/sf/opencsv/opencsv/2.3/opencsv-2.3.jar","file:/C:/Users/tbb/.m2/repository/org/apache/hive/hive-shims/2.3.9/hive-shims-2.3.9.jar","file:/C:/Users/tbb/.m2/repository/org/apache/hive/shims/hive-shims-common/2.3.9/hive-shims-common-2.3.9.jar","file:/C:/Users/tbb/.m2/repository/org/apache/hive/shims/hive-shims-0.23/2.3.9/hive-shims-0.23-2.3.9.jar","file:/C:/Users/tbb/.m2/repository/org/apache/hive/shims/hive-shims-scheduler/2.3.9/hive-shims-scheduler-2.3.9.jar","file:/C:/Users/tbb/.m2/repository/org/apache/hive/hive-llap-common/2.3.9/hive-llap-common-2.3.9.jar","file:/C:/Users/tbb/.m2/repository/org/apache/hive/hive-llap-client/2.3.9/hive-llap-client-2.3.9.jar","file:/C:/Users/tbb/.m2/repository/org/apache/httpcomponents/httpclient/4.5.13/httpclient-4.5.13.jar","file:/C:/Users/tbb/.m2/repository/org/apache/httpcomponents/httpcore/4.4.14/httpcore-4.4.14.jar","file:/C:/Users/tbb/.m2/repository/org/codehaus/jackson/jackson-mapper-asl/1.9.13/jackson-mapper-asl-1.9.13.jar","file:/C:/Users/tbb/.m2/repository/org/codehaus/jackson/jackson-core-asl/1.9.13/jackson-core-asl-1.9.13.jar","file:/C:/Users/tbb/.m2/repository/joda-time/joda-time/2.10.10/joda-time-2.10.10.jar","file:/C:/Users/tbb/.m2/repository/org/jodd/jodd-core/3.5.2/jodd-core-3.5.2.jar","file:/C:/Users/tbb/.m2/repository/org/datanucleus/datanucleus-core/4.1.17/datanucleus-core-4.1.17.jar","file:/C:/Users/tbb/.m2/repository/org/apache/thrift/libthrift/0.12.0/libthrift-0.12.0.jar","file:/C:/Users/tbb/.m2/repository/org/apache/thrift/libfb303/0.9.3/libfb303-0.9.3.jar","file:/C:/Users/tbb/.m2/repository/org/apache/derby/derby/10.14.2.0/derby-10.14.2.0.jar","file:/C:/Users/tbb/.m2/repository/org/apache/spark/spark-catalyst_2.12/3.2.0/spark-catalyst_2.12-3.2.0.jar","file:/C:/Users/tbb/.m2/repository/org/scala-lang/modules/scala-parser-combinators_2.12/1.1.2/scala-parser-combinators_2.12-1.1.2.jar","file:/C:/Users/tbb/.m2/repository/org/codehaus/janino/janino/3.0.16/janino-3.0.16.jar","file:/C:/Users/tbb/.m2/repository/org/codehaus/janino/commons-compiler/3.0.16/commons-compiler-3.0.16.jar","file:/C:/Users/tbb/.m2/repository/org/antlr/antlr4-runtime/4.8/antlr4-runtime-4.8.jar","file:/C:/Users/tbb/.m2/repository/javax/xml/bind/jaxb-api/2.2.11/jaxb-api-2.2.11.jar","file:/C:/Users/tbb/.m2/repository/org/apache/arrow/arrow-vector/2.0.0/arrow-vector-2.0.0.jar","file:/C:/Users/tbb/.m2/repository/org/apache/arrow/arrow-format/2.0.0/arrow-format-2.0.0.jar","file:/C:/Users/tbb/.m2/repository/org/apache/arrow/arrow-memory-core/2.0.0/arrow-memory-core-2.0.0.jar","file:/C:/Users/tbb/.m2/repository/com/google/flatbuffers/flatbuffers-java/1.9.0/flatbuffers-java-1.9.0.jar","file:/C:/Users/tbb/.m2/repository/org/apache/arrow/arrow-memory-netty/2.0.0/arrow-memory-netty-2.0.0.jar","file:/C:/Users/tbb/.m2/repository/io/smartdatalake/spark-extensions_2.12/3.2.0/spark-extensions_2.12-3.2.0.jar","file:/C:/Users/tbb/.m2/repository/com/databricks/spark-xml_2.12/0.14.0/spark-xml_2.12-0.14.0.jar","file:/C:/Users/tbb/.m2/repository/org/glassfish/jaxb/txw2/2.3.4/txw2-2.3.4.jar","file:/C:/Users/tbb/.m2/repository/org/apache/ws/xmlschema/xmlschema-core/2.2.5/xmlschema-core-2.2.5.jar","file:/C:/Users/tbb/.m2/repository/org/apache/spark/spark-avro_2.12/3.2.0/spark-avro_2.12-3.2.0.jar","file:/C:/Users/tbb/.m2/repository/org/tukaani/xz/1.8/xz-1.8.jar","file:/C:/Users/tbb/.m2/repository/org/apache/spark/spark-tags_2.12/3.2.0/spark-tags_2.12-3.2.0.jar","file:/C:/Users/tbb/.m2/repository/commons-io/commons-io/2.8.0/commons-io-2.8.0.jar","file:/C:/Users/tbb/.m2/repository/com/crealytics/spark-excel_2.12/0.14.0/spark-excel_2.12-0.14.0.jar","file:/C:/Users/tbb/.m2/repository/com/norbitltd/spoiwo_2.12/1.8.0/spoiwo_2.12-1.8.0.jar","file:/C:/Users/tbb/.m2/repository/com/github/pjfanning/excel-streaming-reader/2.3.6/excel-streaming-reader-2.3.6.jar","file:/C:/Users/tbb/.m2/repository/xml-apis/xml-apis/1.4.01/xml-apis-1.4.01.jar","file:/C:/Users/tbb/.m2/repository/com/github/pjfanning/poi-shared-strings/1.0.4/poi-shared-strings-1.0.4.jar","file:/C:/Users/tbb/.m2/repository/com/h2database/h2/1.4.200/h2-1.4.200.jar","file:/C:/Users/tbb/.m2/repository/org/apache/commons/commons-compress/1.21/commons-compress-1.21.jar","file:/C:/Users/tbb/.m2/repository/org/apache/poi/poi/4.1.2/poi-4.1.2.jar","file:/C:/Users/tbb/.m2/repository/org/apache/commons/commons-collections4/4.4/commons-collections4-4.4.jar","file:/C:/Users/tbb/.m2/repository/com/zaxxer/SparseBitSet/1.2/SparseBitSet-1.2.jar","file:/C:/Users/tbb/.m2/repository/org/apache/poi/poi-ooxml/4.1.2/poi-ooxml-4.1.2.jar","file:/C:/Users/tbb/.m2/repository/org/apache/poi/poi-ooxml-schemas/4.1.2/poi-ooxml-schemas-4.1.2.jar","file:/C:/Users/tbb/.m2/repository/org/apache/xmlbeans/xmlbeans/3.1.0/xmlbeans-3.1.0.jar","file:/C:/Users/tbb/.m2/repository/com/github/virtuald/curvesapi/1.06/curvesapi-1.06.jar","file:/C:/Users/tbb/.m2/repository/org/hsqldb/hsqldb/2.5.2/hsqldb-2.5.2.jar","file:/C:/Users/tbb/.m2/repository/net/sf/ucanaccess/ucanaccess/5.0.1/ucanaccess-5.0.1.jar","file:/C:/Users/tbb/.m2/repository/com/healthmarketscience/jackcess/jackcess/3.0.1/jackcess-3.0.1.jar","file:/C:/Users/tbb/.m2/repository/com/hierynomus/sshj/0.21.1/sshj-0.21.1.jar","file:/C:/Users/tbb/.m2/repository/org/bouncycastle/bcprov-jdk15on/1.56/bcprov-jdk15on-1.56.jar","file:/C:/Users/tbb/.m2/repository/org/bouncycastle/bcpkix-jdk15on/1.56/bcpkix-jdk15on-1.56.jar","file:/C:/Users/tbb/.m2/repository/com/jcraft/jzlib/1.1.3/jzlib-1.1.3.jar","file:/C:/Users/tbb/.m2/repository/net/i2p/crypto/eddsa/0.3.0/eddsa-0.3.0.jar","file:/C:/Users/tbb/.m2/repository/org/scala-lang/scala-library/2.12.15/scala-library-2.12.15.jar","file:/C:/Users/tbb/.m2/repository/org/scala-lang/scala-reflect/2.12.15/scala-reflect-2.12.15.jar","file:/C:/Users/tbb/.m2/repository/org/scala-lang/scala-compiler/2.12.15/scala-compiler-2.12.15.jar","file:/C:/Users/tbb/.m2/repository/org/scala-lang/modules/scala-xml_2.12/1.2.0/scala-xml_2.12-1.2.0.jar","file:/C:/Users/tbb/.m2/repository/com/typesafe/config/1.4.1/config-1.4.1.jar","file:/C:/Users/tbb/.m2/repository/com/github/kxbmap/configs_2.12/0.6.1/configs_2.12-0.6.1.jar","file:/C:/Users/tbb/.m2/repository/org/scala-lang/modules/scala-collection-compat_2.12/2.4.3/scala-collection-compat_2.12-2.4.3.jar","file:/C:/Users/tbb/.m2/repository/org/scalaj/scalaj-http_2.12/2.3.0/scalaj-http_2.12-2.3.0.jar","file:/C:/Users/tbb/.m2/repository/org/keycloak/keycloak-core/15.0.2/keycloak-core-15.0.2.jar","file:/C:/Users/tbb/.m2/repository/org/keycloak/keycloak-common/15.0.2/keycloak-common-15.0.2.jar","file:/C:/Users/tbb/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar","file:/C:/Users/tbb/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.12.3/jackson-core-2.12.3.jar","file:/C:/Users/tbb/.m2/repository/org/keycloak/keycloak-admin-client/15.0.2/keycloak-admin-client-15.0.2.jar","file:/C:/Users/tbb/.m2/repository/org/jboss/resteasy/resteasy-multipart-provider/3.13.2.Final/resteasy-multipart-provider-3.13.2.Final.jar","file:/C:/Users/tbb/.m2/repository/org/jboss/resteasy/resteasy-jaxrs/3.13.2.Final/resteasy-jaxrs-3.13.2.Final.jar","file:/C:/Users/tbb/.m2/repository/com/github/stephenc/jcip/jcip-annotations/1.0-1/jcip-annotations-1.0-1.jar","file:/C:/Users/tbb/.m2/repository/com/sun/mail/jakarta.mail/1.6.5/jakarta.mail-1.6.5.jar","file:/C:/Users/tbb/.m2/repository/org/apache/james/apache-mime4j/0.6/apache-mime4j-0.6.jar","file:/C:/Users/tbb/.m2/repository/org/jboss/resteasy/resteasy-jackson2-provider/3.13.2.Final/resteasy-jackson2-provider-3.13.2.Final.jar","file:/C:/Users/tbb/.m2/repository/com/fasterxml/jackson/jaxrs/jackson-jaxrs-json-provider/2.10.5/jackson-jaxrs-json-provider-2.10.5.jar","file:/C:/Users/tbb/.m2/repository/com/fasterxml/jackson/jaxrs/jackson-jaxrs-base/2.10.5/jackson-jaxrs-base-2.10.5.jar","file:/C:/Users/tbb/.m2/repository/com/github/fge/json-patch/1.9/json-patch-1.9.jar","file:/C:/Users/tbb/.m2/repository/com/github/fge/jackson-coreutils/1.6/jackson-coreutils-1.6.jar","file:/C:/Users/tbb/.m2/repository/com/github/fge/msg-simple/1.1/msg-simple-1.1.jar","file:/C:/Users/tbb/.m2/repository/com/github/fge/btf/1.2/btf-1.2.jar","file:/C:/Users/tbb/.m2/repository/com/google/guava/guava/14.0.1/guava-14.0.1.jar","file:/C:/Users/tbb/.m2/repository/org/jboss/resteasy/resteasy-jaxb-provider/3.13.2.Final/resteasy-jaxb-provider-3.13.2.Final.jar","file:/C:/Users/tbb/.m2/repository/org/jboss/spec/javax/xml/bind/jboss-jaxb-api_2.3_spec/2.0.0.Final/jboss-jaxb-api_2.3_spec-2.0.0.Final.jar","file:/C:/Users/tbb/.m2/repository/org/glassfish/jaxb/jaxb-runtime/2.3.2/jaxb-runtime-2.3.2.jar","file:/C:/Users/tbb/.m2/repository/com/sun/istack/istack-commons-runtime/3.0.8/istack-commons-runtime-3.0.8.jar","file:/C:/Users/tbb/.m2/repository/org/jboss/resteasy/resteasy-client/4.5.7.Final/resteasy-client-4.5.7.Final.jar","file:/C:/Users/tbb/.m2/repository/org/jboss/resteasy/resteasy-client-api/4.5.7.Final/resteasy-client-api-4.5.7.Final.jar","file:/C:/Users/tbb/.m2/repository/org/jboss/resteasy/resteasy-core-spi/4.5.7.Final/resteasy-core-spi-4.5.7.Final.jar","file:/C:/Users/tbb/.m2/repository/org/jboss/spec/javax/annotation/jboss-annotations-api_1.3_spec/2.0.1.Final/jboss-annotations-api_1.3_spec-2.0.1.Final.jar","file:/C:/Users/tbb/.m2/repository/org/jboss/resteasy/resteasy-core/4.5.7.Final/resteasy-core-4.5.7.Final.jar","file:/C:/Users/tbb/.m2/repository/com/ibm/async/asyncutil/0.1.0/asyncutil-0.1.0.jar","file:/C:/Users/tbb/.m2/repository/org/eclipse/microprofile/config/microprofile-config-api/1.4/microprofile-config-api-1.4.jar","file:/C:/Users/tbb/.m2/repository/io/smallrye/config/smallrye-config/1.6.1/smallrye-config-1.6.1.jar","file:/C:/Users/tbb/.m2/repository/io/smallrye/config/smallrye-config-common/1.6.1/smallrye-config-common-1.6.1.jar","file:/C:/Users/tbb/.m2/repository/org/jboss/logging/jboss-logging/3.3.2.Final/jboss-logging-3.3.2.Final.jar","file:/C:/Users/tbb/.m2/repository/org/jboss/spec/javax/ws/rs/jboss-jaxrs-api_2.1_spec/2.0.1.Final/jboss-jaxrs-api_2.1_spec-2.0.1.Final.jar","file:/C:/Users/tbb/.m2/repository/io/monix/monix-eval_2.12/3.4.0/monix-eval_2.12-3.4.0.jar","file:/C:/Users/tbb/.m2/repository/io/monix/monix-catnap_2.12/3.4.0/monix-catnap_2.12-3.4.0.jar","file:/C:/Users/tbb/.m2/repository/org/typelevel/cats-effect_2.12/2.5.1/cats-effect_2.12-2.5.1.jar","file:/C:/Users/tbb/.m2/repository/io/monix/monix-execution_2.12/3.4.0/monix-execution_2.12-3.4.0.jar","file:/C:/Users/tbb/.m2/repository/io/monix/monix-internal-jctools_2.12/3.4.0/monix-internal-jctools_2.12-3.4.0.jar","file:/C:/Users/tbb/.m2/repository/io/monix/implicitbox_2.12/0.3.4/implicitbox_2.12-0.3.4.jar","file:/C:/Users/tbb/.m2/repository/org/reactivestreams/reactive-streams/1.0.3/reactive-streams-1.0.3.jar","file:/C:/Users/tbb/.m2/repository/com/github/mutcianm/ascii-graphs_2.12/0.0.6/ascii-graphs_2.12-0.0.6.jar","file:/C:/Users/tbb/.m2/repository/org/apache/commons/commons-pool2/2.8.0/commons-pool2-2.8.0.jar","file:/C:/Users/tbb/.m2/repository/org/apache/tika/tika-core/1.24.1/tika-core-1.24.1.jar","file:/C:/Users/tbb/.m2/repository/org/reflections/reflections/0.9.12/reflections-0.9.12.jar","file:/C:/Users/tbb/.m2/repository/org/javassist/javassist/3.26.0-GA/javassist-3.26.0-GA.jar","file:/C:/Users/tbb/.m2/repository/com/github/takezoe/runtime-scaladoc-reader_2.12/1.0.2/runtime-scaladoc-reader_2.12-1.0.2.jar","file:/C:/Users/tbb/.m2/repository/org/json4s/json4s-ext_2.11/3.7.0-M11/json4s-ext_2.11-3.7.0-M11.jar","file:/C:/Users/tbb/.m2/repository/org/joda/joda-convert/2.2.1/joda-convert-2.2.1.jar","file:/C:/Users/tbb/.m2/repository/com/networknt/json-schema-validator/1.0.42/json-schema-validator-1.0.42.jar","file:/C:/Users/tbb/.m2/repository/org/jruby/joni/joni/2.1.31/joni-2.1.31.jar","file:/C:/Users/tbb/.m2/repository/org/jruby/jcodings/jcodings/1.0.46/jcodings-1.0.46.jar","file:/C:/Users/tbb/.m2/repository/com/fasterxml/jackson/datatype/jackson-datatype-jsr310/2.12.3/jackson-datatype-jsr310-2.12.3.jar","file:/C:/Users/tbb/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.12.3/jackson-annotations-2.12.3.jar","file:/C:/Users/tbb/.m2/repository/log4j/log4j/1.2.17/log4j-1.2.17.jar","file:/C:/Users/tbb/.m2/repository/org/typelevel/cats-core_2.12/1.5.0/cats-core_2.12-1.5.0.jar","file:/C:/Users/tbb/.m2/repository/org/typelevel/cats-macros_2.12/1.5.0/cats-macros_2.12-1.5.0.jar","file:/C:/Users/tbb/.m2/repository/org/typelevel/cats-kernel_2.12/1.5.0/cats-kernel_2.12-1.5.0.jar","file:/C:/Users/tbb/.m2/repository/org/typelevel/machinist_2.12/0.6.6/machinist_2.12-0.6.6.jar","file:/C:/Users/tbb/.m2/repository/org/json4s/json4s-ext_2.12/3.7.0-M11/json4s-ext_2.12-3.7.0-M11.jar","file:/C:/Users/tbb/.m2/repository/org/eclipse/jetty/jetty-servlet/9.4.43.v20210629/jetty-servlet-9.4.43.v20210629.jar","file:/C:/Users/tbb/.m2/repository/org/eclipse/jetty/jetty-security/9.4.43.v20210629/jetty-security-9.4.43.v20210629.jar","file:/C:/Users/tbb/.m2/repository/org/eclipse/jetty/jetty-server/9.4.43.v20210629/jetty-server-9.4.43.v20210629.jar","file:/C:/Users/tbb/.m2/repository/javax/servlet/javax.servlet-api/3.1.0/javax.servlet-api-3.1.0.jar","file:/C:/Users/tbb/.m2/repository/org/eclipse/jetty/jetty-util-ajax/9.4.43.v20210629/jetty-util-ajax-9.4.43.v20210629.jar","file:/C:/Users/tbb/.m2/repository/org/eclipse/jetty/jetty-http/9.4.43.v20210629/jetty-http-9.4.43.v20210629.jar","file:/C:/Users/tbb/.m2/repository/org/eclipse/jetty/jetty-util/9.4.43.v20210629/jetty-util-9.4.43.v20210629.jar","file:/C:/Users/tbb/.m2/repository/org/eclipse/jetty/jetty-io/9.4.43.v20210629/jetty-io-9.4.43.v20210629.jar","file:/C:/Users/tbb/.m2/repository/org/glassfish/jersey/media/jersey-media-json-jackson/2.34/jersey-media-json-jackson-2.34.jar","file:/C:/Users/tbb/.m2/repository/org/glassfish/jersey/ext/jersey-entity-filtering/2.34/jersey-entity-filtering-2.34.jar","file:/C:/Users/tbb/.m2/repository/com/fasterxml/jackson/module/jackson-module-jaxb-annotations/2.12.3/jackson-module-jaxb-annotations-2.12.3.jar","file:/C:/Users/tbb/.m2/repository/jakarta/xml/bind/jakarta.xml.bind-api/2.3.2/jakarta.xml.bind-api-2.3.2.jar","file:/C:/Users/tbb/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.1/jakarta.activation-api-1.2.1.jar","file:/C:/Program%20Files/JetBrains/IntelliJ%20IDEA%202019.2.4/lib/idea_rt.jar"]}}},"_sparkSession":{"sessionUUID":"bb98bad9-c5f4-4014-bda9-887e4d6692e7","traceEnabled":false},"referenceTimestampOrNow":[2022,2,7,14,58,37,62000000]}`

Reply:
You are right, thats very noisy! Lets make it transient. As far as i remember we dont serialize the context for executors, and it should be ok to make serializableHadoopConf transient. Can you add a comment about why this is transient, please.

Reply:
done